src/TernaryLib.sol

```diff
@@ -26,14 +26,44 @@ library TernaryLib {
         }
     }
 
+    /// @notice Equivalent to: `uint256(x < 0 ? -x : x)`
+    function abs(int256 x) internal pure returns (uint256 y) {
+        assembly {
+            // mask = 0 if x >= 0 else -1
+            let mask := sub(0, slt(x, 0))
+            // If x >= 0, |x| = x = 0 ^ x
+            // If x < 0, |x| = ~~|x| = ~(-|x| - 1) = ~(x - 1) = -1 ^ (x - 1)
+            // Either case, |x| = mask ^ (x + mask)
+            y := xor(mask, add(mask, x))
+        }
+    }
+
     /// @notice Equivalent to: `a < b ? a : b`
-    function min(uint256 a, uint256 b) internal pure returns (uint256) {
-        return ternary(a < b, a, b);
+    function min(uint256 a, uint256 b) internal pure returns (uint256 res) {
+        assembly {
+            res := xor(b, mul(xor(a, b), lt(a, b)))
+        }
     }
 
     /// @notice Equivalent to: `a > b ? a : b`
-    function max(uint256 a, uint256 b) internal pure returns (uint256) {
-        return ternary(a > b, a, b);
+    function max(uint256 a, uint256 b) internal pure returns (uint256 res) {
+        assembly {
+            res := xor(b, mul(xor(a, b), gt(a, b)))
+        }
+    }
+
+    /// @notice Equivalent to: `condition ? (b, a) : (a, b)`
+    function switchIf(
+        bool condition,
+        uint256 a,
+        uint256 b
+    ) internal pure returns (uint256, uint256) {
+        assembly {
+            let diff := mul(xor(a, b), condition)
+            a := xor(a, diff)
+            b := xor(b, diff)
+        }
+        return (a, b);
```

Comment:
Please double check return type function overloading. I don't think Solidity support this type of overloading. At least C++ does not.